tensorflow/c/kernels/summary_op_benchmark_test.cc

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <string>
+
+#include "tensorflow/core/common_runtime/kernel_benchmark_testlib.h"
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/graph/node_builder.h"
+#include "tensorflow/core/platform/test.h"
+#include "tensorflow/core/platform/test_benchmark.h"
+#include "tensorflow/core/framework/tensor_shape.h"
+
+namespace tensorflow {
+namespace {
+
+Graph* BM_ScalarSummaryOp(TensorShape shape, std::string tag, 
+                          float value) {
+  Graph* g = new Graph(OpRegistry::Global());
+  Tensor tags(DT_STRING, shape);
+  Tensor values(DT_FLOAT, shape);
+  for (int i = 0; i < tags.NumElements(); ++i){ 
+  	tags.flat<tstring>()(i) = tag; 
+  	values.flat<float>()(i) = value; 
+  } 
+  Node* ret;
+  TF_CHECK_OK(NodeBuilder(g->NewName("dummy"), "ScalarSummary")
+                  .Input(test::graph::Constant(g, tags))
+                  .Input(test::graph::Constant(g, values))
+                  .Attr("T", DT_FLOAT)
+                  .Finalize(g, &ret));
+  return g;
+}
+
+// Macro used to parse initializer list for tensorshape 
+#define DIMARGS(...) {__VA_ARGS__}
+// Random parameters for testing
+constexpr char longTagParam[] = "LONGTAG____________________________"; 
+constexpr float largeValueParam = 2352352.2623433; 
+
+#define BM_ScalarSummaryDev(device, dims, name, tag, value)       		\
+	void BM_ScalarSummary##name##device(int iters) { 	                  \
```

Comment:
Some parts of this macro are not well-aligned.
For e.g. function body should be indented by 2 spaces relative to function header. Also, I can't tell if it

Also, can you call `testing::StopTiming();` at the start of this function and testing::SartTiming() right before `test::Benchmark` call? Basically, we just want to measure how long it takes to run `test::Benchmark`. That's why we want to stop timing as soon as we can and start it right before `test::Benchmark` call.